src/hotspot/share/opto/escape.cpp

```diff
@@ -3149,7 +3166,7 @@ void ConnectionGraph::split_unique_types(GrowableArray<Node *>  &alloc_worklist,
         // so it could be eliminated.
         alloc->as_Allocate()->_is_scalar_replaceable = true;
       }
-      set_escape_state(ptnode_adr(n->_idx), es); // CheckCastPP escape state
+      set_escape_state(ptnode_adr(n->_idx), es NOT_PRODUCT(COMMA "reason unknown (2)")); // CheckCastPP escape state
```

Comment:
Escape state of allocation is assigned to CheckCastPP node. `es` value should be `NoEscape` - see line 3109.
May be reason should be "assign state of related Allocation".